GameServerLib/Items/Inventory.cs

```diff
@@ -133,24 +134,14 @@ public void RemoveItem(byte slot, IObjAiBase owner, int stacksToRemove = 1)
                     {
                         owner.Stats.RemoveModifier(ItemScripts[itemID].StatsModifier);
                     }
-                    ItemScripts.Remove(GetItem(slot).ItemData.ItemId);
+                    ItemScripts.Remove(itemID);
                 }
                 Items[slot] = null;
             }
             else
             {
                 Items[slot].SetStacks(finalStacks);
             }
-
-            if (!HasItemWithID(itemID) && owner != null)
```

Comment:
Revert deletion and change condition to
`if (finalStacks <= 0 || !HasItemWithID(itemID)) && owner != null && ItemScripts.ContainsKey(itemID)`.

Reply:
Done